Shippo/Track.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using Newtonsoft.Json;
+
+namespace Shippo {
+    [JsonObject (MemberSerialization.OptIn)]
+    public class Track : ShippoId {
+        [JsonProperty (PropertyName = "carrier")]
+        public object Carrier { get; set; }
```

Comment:
Do you mean to say that `JsonProperty` needs the setters?  If so, then its good. Interestingly Java's `GSON` doesn't need the setters. It is able to set private variables using Java reflection. 

I also just realized that all the members are public. They should probably be private with getter/setter. The getter/setter are not required if they are public.